templates/common/exec.go

```diff
@@ -39,9 +39,11 @@ const DefaultRunTimeout = time.Minute
 // If the incoming context doesn't already have a timeout, then a default
 // timeout will be added (see DefaultRunTimeout).
 //
+// exitCode is -1 if the command could not be executed.
+//
 // If the command fails, the error message will include the contents of stdout
 // and stderr. This saves boilerplate in the caller.
-func Run(ctx context.Context, args ...string) (stdout, stderr string, _ error) {
+func Run(ctx context.Context, args ...string) (stdout, stderr string, exitCode int, _ error) {
```

Comment:
I'm not sure about this approach. I think it would be better to implement our own error type that embeds the exit code in the returned error. I also see `RunAllowNonZero` added, but it doesn't appear to be used anywhere?

Reply:
Removed, looks like this was a vestigial bit of code that wasn't used in the final revision.